hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/Client.java

```diff
@@ -455,6 +385,10 @@ private class Connection extends Thread {
         Consumer<Connection> removeMethod) {
       this.remoteId = remoteId;
       this.server = remoteId.getAddress();
+      this.rpcRequestThread = new Thread(new RpcRequestSender(),
+          "IPC Parameter Sending Thread for " + remoteId);
+      this.rpcRequestThread.setDaemon(true);
+      this.rpcRequestThread.start();
```

Comment:
Maybe we should start `rpcRequestThread`  after `setupConnection()`, because if we started it here, the socket might not be available.

Reply:
Yeah, I moved the thread start to the top of the run method in the Connection thread.